lib/fb-jwt-client.js

```diff
@@ -165,10 +170,16 @@ class FBJWTClient {
    * Access token
    *
    **/
-  generateAccessToken (data) {
+  generateAccessToken (data, secret, algorithm) {
     const checksum = crypto.createHash('sha256').update(JSON.stringify(data)).digest('hex')
 
-    return jwt.sign({checksum}, this.serviceToken, {algorithm})
+    debugger
+
+    if (secret === undefined) {
```

Comment:
The default return from a function is `undefined` so it would be simpler to write:
```
   if (secret) {
     return jwt.sign({checksum}, secret, {issuer: this.serviceSlug, algorithm: algorithm})
   }
```
Which also ensures that `secret` is truthy (it's defined, with a value) which is generally better and almost certainly what you want

Reply:
If `secret` and `algorithm` must have a value, then:

```
  if (secret && algorithm) {
     return jwt.sign({checksum}, secret, {issuer: this.serviceSlug, algorithm: algorithm})
   }
```

Reply:
You could make `algorithm` optional by defaulting the argument, like so:
```
  generateAccessToken (data, secret, algorithm = 'HS256') { /* algorithm has a default value */
    const checksum = crypto.createHash('sha256').update(JSON.stringify(data)).digest('hex')

    if (secret) {
      return jwt.sign({checksum}, secret, {issuer: this.serviceSlug, algorithm: algorithm})
    }
  }

Reply:
made a slight tweak